### PR TITLE
CPM-634: Add Uuids in IdentifierResult from PQB

### PR DESCRIPTION
**Pull request by Balamung**
**Description (for Contributor and Core Developer)**

When done, notify octopus team about updating components/catalogs/back/src/Infrastructure/TemporaryEnrichmentBridge/SearchAfterSizeUuidResultCursorFactory.php#L57 to also add the id in _source of the esQuery

**Definition Of Done (for Core Developer only)**

- [x] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/IdentifierResult.php [hunk 1]
```diff
@@ -42,8 +45,14 @@ public function __construct(string $identifier, string $type)
             );
         }
 
+
```
Comment (pierallard) [inline src/Akeneo/Pim/Enrichment/Bundle/Elasticsearch/IdentifierResult.php:48]: 2 empty lines :fouet:

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/FromSizeIdentifierResultCursorFactorySpec.php [hunk 2]
```diff
@@ -39,16 +43,16 @@ function it_creates_a_product_identifier_cursor($esClient)
         $result = ['hits' => [
             'total' => ['value' => 42, 'relation' => 'eq'],
             'hits'  => [
-                ['_source' => ['identifier' => 'product_1', 'document_type' => ProductInterface::class]],
-                ['_source' => ['identifier' => 'product_model_2', 'document_type' => ProductModelInterface::class]],
+                ['_source' => ['identifier' => 'product_1', 'document_type' => ProductInterface::class, 'id' => $uuid1->toString()]],
+                ['_source' => ['identifier' => 'product_model_2', 'document_type' => ProductModelInterface::class, 'id' => $uuid2->toString()]],
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/FromSizeIdentifierResultCursorFactorySpec.php:47]: Please use real data
product_1 should be product_a_uuid with the same 'id'
and product_model_2 id should be 2

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/FromSizeIdentifierResultCursorFactorySpec.php [hunk 3]
```diff
@@ -57,8 +61,8 @@ function it_creates_a_product_identifier_cursor($esClient)
 
         $this->createCursor($esQuery, $options)->shouldBeLike(new IdentifierResultCursor(
             [
-                new IdentifierResult('product_1', ProductInterface::class),
-                new IdentifierResult('product_model_2', ProductModelInterface::class),
+                new IdentifierResult('product_1', ProductInterface::class, $uuid1->toString()),
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/FromSizeIdentifierResultCursorFactorySpec.php:64]: same here

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/CursorSpec.php [hunk 4]
```diff
@@ -35,11 +36,11 @@ function let(
                     'total' => ['value' => 4, 'relation' => 'eq'],
                     'hits' => [
                         [
-                            '_source' => ['identifier' => 'a-variant-product', 'document_type' => ProductInterface::class],
+                            '_source' => ['identifier' => 'a-variant-product', 'document_type' => ProductInterface::class, 'id' => 'product_' . $variantProduct->getUuid()->toString()],
                             'sort' => ['#a-variant-product']
                         ],
                         [
-                            '_source' => ['identifier' => 'a-sub-product-model', 'document_type' => ProductModelInterface::class],
+                            '_source' => ['identifier' => 'a-sub-product-model', 'document_type' => ProductModelInterface::class, 'id' => 'a-sub-product-model'],
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/CursorSpec.php:43]: id should be "product_model_a-sub-product-model"

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/CursorSpec.php [hunk 5]
```diff
@@ -94,11 +97,11 @@ function it_is_iterable(
                     'total' => ['value' => 4],
                     'hits' => [
                         [
-                            '_source' => ['identifier' => 'a-root-product-model', 'document_type' => ProductModelInterface::class],
+                            '_source' => ['identifier' => 'a-root-product-model', 'document_type' => ProductModelInterface::class, 'id' => 'a-root-product-model'],
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/CursorSpec.php:100]: Same here

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/CursorSpec.php [hunk 6]
```diff
@@ -169,11 +175,11 @@ function it_is_iterable_with_products_and_product_models_having_the_same_identif
                     'total' => ['value' => 4],
                     'hits' => [
                         [
-                            '_source' => ['identifier' => 'foo', 'document_type' => ProductModelInterface::class],
+                            '_source' => ['identifier' => 'foo', 'document_type' => ProductModelInterface::class, 'id' => 'foo'],
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/CursorSpec.php:178]: product_model_foo

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/IdentifierResultSpec.php [hunk 7]
```diff
@@ -5,42 +5,50 @@
...
     {
-        $this->beConstructedWith('foo', ProductInterface::class);
+        $this->beConstructedWith('foo', ProductInterface::class, Uuid::uuid4()->toString());
         $this->isProductIdentifierEquals('foo')->shouldReturn(true);
     }
 
     function it_determines_if_not_equals_to_a_product_identifier()
     {
-        $this->beConstructedWith('foo', ProductModelInterface::class);
+        $this->beConstructedWith('foo', ProductModelInterface::class, 'foo');
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/IdentifierResultSpec.php:39]: Same

File: tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/SearchAfterSizeIdentifierResultCursorFactorySpec.php [hunk 8]
```diff
@@ -40,25 +43,25 @@ function it_creates_a_product_identifier_cursor($esClient)
         $result = ['hits' => [
             'total' => ['value' => 42, 'relation' => 'eq'],
             'hits'  => [
-                ['_source' => ['identifier' => 'product_1', 'document_type' => ProductInterface::class]],
-                ['_source' => ['identifier' => 'product_model_2', 'document_type' => ProductModelInterface::class]],
+                ['_source' => ['identifier' => 'product_1', 'document_type' => ProductInterface::class, 'id' => 'product_' . $uuid->toString()]],
+                ['_source' => ['identifier' => 'product_model_2', 'document_type' => ProductModelInterface::class, 'id' => 'product_model_2']],
```
Comment (pierallard) [inline tests/back/Pim/Enrichment/Specification/Bundle/Elasticsearch/SearchAfterSizeIdentifierResultCursorFactorySpec.php:47]: product_model_product_model_2

**Review by pierallard — COMMENTED** [review]
Please check each time you return a product model id (I didn't check every class), you should have an id like "product_model_itscode"

**Review by RiffVentura — APPROVED** [review]

**Review by pierallard — APPROVED** [review]

**Pull request merged**